nhs_context_logging/logger.py

```diff
@@ -582,6 +590,32 @@ def internal_id(self) -> str:
     def log_reference(self):
         return self.fields.get(Constants.LOG_REFERENCE_FIELD)
 
+    async def async_handle_result(self, func, *args, **inner_kwargs):
+        try:
+            res = await func(*args, **inner_kwargs)
+        except Exception as error:
+            await self.async_add_fields(**{Constants.ACTION_RESULT: error})
+            raise error
+        else:
+            await self.async_add_fields(**{Constants.ACTION_RESULT: res})
+            return res
+
+    def handle_result(self, func, *args, **inner_kwargs):
+        try:
+            res = func(*args, **inner_kwargs)
+        except Exception as error:
+            self.add_fields(**{Constants.ACTION_RESULT: error})
```

Comment:
why not check `self.log_result` here  rather than adding and popping below?

Reply:
i've removed the adding/popping procedure. much simpler now.

also i've de-nested by removing the error handling and so my concern about branching isn't an issue now.